crates/polars-core/src/serde/series.rs

```diff
@@ -230,6 +227,15 @@ impl<'de> Deserialize<'de> for Series {
                     dt => {
                         panic!("{dt:?} dtype deserialization not yet implemented")
                     }
+                }?;
+
+                if let Some(f) = bit_settings {
```

Comment:
We can write this a little more functional:


```rust
bit_settings.map_or_else(
    |_| Ok(s),
    |s| s.set_flags(f)
          .map_err(|_| de::Error::custom("bit flags are corrupt"))) // custom of no capitals in error msg
);

```

Reply:
This would not work AFAIK as the `Ok(s)` moves the value and you can not reuse it again. Also `set_flags()` does not return the series so we would need to add an extra statement.

Happy to convert to something more idiomatic but can not come up with it

Reply:
Ah, right. Then we can at least do the `map_err` part.